src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -471,7 +476,9 @@ private void parseDoc(Map<String, String> namespacePrefixesByUri, String lastSav
                 final String instanceId = instanceNodeIdStrs.get(instanceIndex);
                 final String instanceSrc = parseInstanceSrc(instance, lastSavedSrc);
 
-                if (instanceSrc != null) {
+                // Only read in an external instance if its ID is used in the primary instance as an argument to an
+                // instance() call
+                if (instanceSrc != null && referencedInstanceIds.contains(instanceId)) {
```

Comment:
At this point, you could assert that `fruits` is not part of the `referencedInstanceIds` in the first test